src/plugins/intel_gna/transformations/unfuse_reshape_and_transpose.hpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (C) 2021 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#pragma once
+
+#include <ngraph/pass/graph_rewrite.hpp>
+
+namespace GNAPluginNS {
+
+/**
+ * @brief Replace 2d->4d reshape to pair of 2 reshapes (before Convolution)
```

Comment:
@elilobanova In the theory, we can extend this transformation for 3d cases. But this PR is mostly about new kaldi IRs. For kaldi we have 2d inputs only (batch dimension is added by MO to the flatten input). Maybe somewhere we can meet NxHxW -> Nx1xHxW as well. I propose to add support 3d -> 4d when it would be actually needed.